contracts/koh/contracts/KingAutomaton.sol

```diff
@@ -392,84 +550,83 @@ contract KingAutomaton {
 
   // In case the contributor is inactive anyone could call the function AFTER all periods are passed and the funds locked
   // in the proposal address will be returned to treasury. Rejecting the proposal will have the same effect.
-
-  function claimReward(uint256 _id, uint256 _budget) public validBallotBoxID(_id) {
-    updateProposalState(_id);
-    Proposal storage p = proposals[_id];
-    require(p.state == ProposalState.Accepted || p.state == ProposalState.Contested, "Incorrect proposal state!");
-    uint256 paymentDate = p.nextPaymentDate;
-    uint256 remainingPeriods = p.remainingPeriods;
-    uint256 periodLen = p.budgetPeriodLen;
-    uint256 budgetPerPeriod = p.budgetPerPeriod;
-    address proposalAddress = address(_id);
-
-    require(_budget <= budgetPerPeriod, "Budget exceeded!");
-
-    if (paymentDate > 0) {
-      if (paymentDate <= now) {
-        if (paymentDate + periodLen < now) {
-          uint256 missedPeriods = (now - paymentDate) / periodLen;
-          transferInternal(proposalAddress, treasuryAddress, missedPeriods * p.budgetPerPeriod);
-          remainingPeriods -= missedPeriods;
-          paymentDate += periodLen * missedPeriods;
+function claimReward(uint256 _id, uint256 _budget) public validBallotBoxID(_id) {
```

Comment:
Same here - would be good to extract some of this code into the Proposals library.